src/pz390.java

```diff
@@ -2788,9 +2790,11 @@ private void ins_lt_c0() {
 		 */
 		switch (opcode1) {
 		case 0x80: // 1530 "8000" "SSM" "S"
+			set_psw_check(psw_pic_priv); // RPI 1622 DK Privileged operation exception
 			ins_setup_s();
 			break;
```

Comment:
Shouldn't the code check to see if in supervisor state (PSW bit 15 is 0) before the "set_psw_check(psw_pic_priv);" ?

Yes, today's pz390.java doesn't properly build the PSW to reflect problem/supervisor state. It should. I'm willing to look into this. It should be a simple change -- always set PSW bit 15 to 1 (problem state).

Suppose a user adds a new SVC to sz390.java that, when invoked, switches to supervisor state. (Their own MODESET macro that anyone can successfully invoke.) At that point, they should be able to use the privileged instructions.

Of course, this check would be needed at all the other places below where psw_pic_priv is set.

Reply:
Technically, you are right. Consequentially, we should then also change these opcodes from no-ops to really operational opcodes in case the PSW is on. That's quite a challenge. I would be okay with adding the supervisor state check but will not build all code required to make all privileged instructions run as described in the PoP. Quite a few of those cannot even be implemented in the z390 environment. Because we have no address spaces, page tables, etc.
If we add the supervisor bit check but leave the instructions as a no-op for the supervisor state case, would that be good enough for the time being?

Reply:
Suggestion.
1. Create new submodule, say verify_supervisor_state. Could be something like
```
void verify_supervisor_state() {
    // TODO: add check of PSW to determine whether supervisor state
    set_psw_check(psw_pic_priv);
}
```
2. Replace all the current set_psw_check(psw_pic_priv) invocations with verify_supervisor_state().
3. At this point, the code is logically equivalent to what exists now.
4. At some future date, implement the TODO.

This greatly reduces the burden on whoever maintains the code.

Reply:
Good suggestion! I'll give it a try later today or tomorrow...